mcbackend/core.py

```diff
@@ -181,16 +182,20 @@ def to_inferencedata(self, **kwargs) -> InferenceData:
         chain_lengths = {c.cid: len(c) for c in chains}
         if len(set(chain_lengths.values())) != 1:
             _log.warning("Chains vary in length. Lenghts are: %s", chain_lengths)
-
+        
+        if equalize_chain_lengths:
+            # A minimum chain length is introduced so that all chains have equal length
+            clen = min(chain_lengths.values())
         # Aggregate draws and stats, while splitting into warmup/posterior
         warmup_posterior = collections.defaultdict(list)
         warmup_sample_stats = collections.defaultdict(list)
         posterior = collections.defaultdict(list)
         sample_stats = collections.defaultdict(list)
         for c, chain in enumerate(chains):
+            if not equalize_chain_lengths:
```

Comment:
```suggestion
            if clen is None:
```
in combination with initializing it as `None`, this makes it more robust against future changes.